futures-util/src/future/flatten_stream.rs

```diff
@@ -30,11 +33,25 @@ impl<Fut> fmt::Debug for FlattenStream<Fut>
 }
 
 #[derive(Debug)]
-enum State<Fut: Future> {
+enum State<Fut, St> {
     // future is not yet called or called and not ready
     Future(Fut),
     // future resolved to Stream
-    Stream(Fut::Output),
+    Stream(St),
+}
+
+impl<Fut, St> State<Fut, St> {
+    fn get_pin_mut<'a>(self: Pin<&'a mut Self>) -> State<Pin<&'a mut Fut>, Pin<&'a mut St>> {
```

Comment:
`'a` can be removed when https://github.com/rust-lang/rust/pull/60944 is merged.